feathr_project/feathr/spark_provider/_synapse_submission.py

```diff
@@ -326,7 +326,7 @@ def get_driver_log(self, job_id) -> str:
         # @see: https://docs.microsoft.com/en-us/azure/synapse-analytics/spark/connect-monitor-azure-synapse-spark-application-level-metrics
         app_id = self.get_spark_batch_job(job_id).app_id
         url = "%s/sparkhistory/api/v1/sparkpools/%s/livyid/%s/applications/%s/driverlog/stdout/?isDownload=true" % (self._synapse_dev_url, self._spark_pool_name, job_id, app_id)
-        token = self._credential.get_token("https://dev.azuresynapse.net/.default")
+        token = self._credential.get_token("https://dev.azuresynapse.net/.default").token
```

Comment:
>get_token("https://dev.azuresynapse.net/.default") [](http://example.com/codeflow?start=33&length=50)

Synapse token is for protecting Synapse resources only. Please change the scope to [https://management.azure.com/.default](https://management.azure.com/.default) which is the URI for ARM(azure resource management) which is more appreciable here

Reply:
@t-curiekim tried with grap.azure.com url earlier and that didn't work as that as the same net effect as just called .default. Agree, let's try the management.azure.com URL and that might work.

Reply:
Have used the management azure url and seemed to work before. I will try testing a few more times with the management url to confirm.

Reply:
Thank you for testing! I have tried running again now, and works on my side as well 👍